php-mode.el

```diff
@@ -1734,6 +1734,13 @@ The output will appear in the buffer *PHP*."
 (dolist (pattern '("\\.php[s345t]?\\'" "/\\.php_cs\\(\\.dist\\)?\\'" "\\.phtml\\'" "/Amkfile\\'" "\\.amk\\'"))
   (add-to-list 'auto-mode-alist `(,pattern . php-mode) t))
 
+;;;###autoload
+(progn
+  ;; matches `/config.php'
+  ;; refs https://github.com/ejmr/php-mode/pull/362#issuecomment-312404436
+  ;; To avoid conflict with default `conf-maybe-mode' in auto-mode-alist, add it explicitly.
+  (add-to-list 'auto-mode-alist '("[/.]c\\(?:on\\)?f\\(?:i?g\\)?\\(?:\\.[a-zA-Z0-9._-]+\\)?\\.php[s345t]?\\'" . php-mode)))
+
```

Comment:
Does this also match `cfg.php`?  In my opinion this is an example of a regular expression becoming too complex; we have multiple regular expressions like this which I believe [we should gradually replace](https://github.com/ejmr/php-mode/issues/354).  Even though it would be more verbose, we can and I think should rewrite this regular expression now to use `(rx ...)` so that it will be easier to understand exactly what we are matching.

Alternatively, or even together with `(rx ...)`, I think it is better to use a separate call to `add-to-list` to cover the situation of `config.php`.  Having a separate call like `(add-to-list 'auto-mode-alist '("regex for config.php" . php-mode))` will create a tiny amount of code duplication---we don't **need** two calls to `add-to-list`---but in this scenario I believe it helps readability and maintainability.